baremetal_network_provisioning/ml2/extensions/bnp_switchport.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2016 OpenStack Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import webob.exc
+
+from neutron.api import extensions
+from neutron.api.v2 import base
+from neutron.api.v2 import resource
+from neutron import wsgi
+
+from baremetal_network_provisioning.db import bm_nw_provision_db as db
+
+RESOURCE_ATTRIBUTE_MAP = {
+    'bnp-switch-ports': {
+        'switch_name': {'allow_post': False, 'allow_put': False,
+                        'is_visible': True},
+        'neutron_port_id': {'allow_post': False, 'allow_put': False,
+                            'is_visible': True},
+        'switch_port_name': {'allow_post': False, 'allow_put': False,
+                             'is_visible': True},
+        'segmentation_id': {'allow_post': False, 'allow_put': False,
+                            'is_visible': True},
+        'lag_id': {'allow_post': False, 'allow_put': False,
+                   'is_visible': True},
+        'bind_status': {'allow_post': False, 'allow_put': False,
+                        'is_visible': True},
+        'access_type': {'allow_post': False, 'allow_put': False,
+                        'is_visible': True}
+
+    },
+}
+
+
+class BNPSwitchPortController(wsgi.Controller):
+
+    """WSGI Controller for the extension bnp-switch-port."""
+
+    def index(self, request, **kwargs):
+        context = request.context
+        req_dict = dict(request.GET)
+        if req_dict and req_dict.get('fields', None):
+            req_dict.pop('fields')
+        filters = req_dict
+        port_maps = db.get_all_bnp_switch_port_maps(context, **filters)
+        port_list = []
+        for port_map in port_maps:
+            port_dict = {'neutron_port_id': port_map[0],
+                         'switch_port_name': port_map[1],
+                         'lag_id': port_map[2],
+                         'segmentation_id': str(port_map[3]),
+                         'access_type': port_map[4],
+                         'bind_status': str(port_map[5]),
```

Comment:
bind_status : check for 0 and 1
if value is 0 then it is success 
else Failure
Right now raw values being send to REST .